pkg/cmd/delete_credential.go

```
@@ -46,49 +56,32 @@ func NewDeleteCredentialCmd(
 		Use:       "credential",
 		Short:     "Delete credential",
 		Long:      `Delete credential from current env`,
-		RunE:      RunFuncE(s.runStdin(), s.runPrompt()),
+		RunE:      RunFuncE(s.runStdin(), s.runFormula()),
 		ValidArgs: []string{""},
 		Args:      cobra.OnlyValidArgs,
 	}
-	cmd.LocalFlags()
+	flags := cmd.Flags()
+	flags.String(providerFlagName, "", providerFlagDescription)
+
 	return cmd
 }
 
-func (d deleteCredentialCmd) runPrompt() CommandRunnerFunc {
+func (d deleteCredentialCmd) runFormula() CommandRunnerFunc {
 	return func(cmd *cobra.Command, args []string) error {
-		env, err := d.currentEnv()
+		curEnv, err := d.currentEnv()
 		if err != nil {
 			return err
 		}
-		prompt.Info(fmt.Sprintf("Current env: %s", env))
+		prompt.Info(fmt.Sprintf("Current env: %s", curEnv))
 
-		data, err := d.ReadCredentialsValueInEnv(d.CredentialsPath(), env)
+		inputParams, err := d.resolveInput(cmd, curEnv)
 		if err != nil {
 			return err
-		}
-
-		if len(data) <= 0 {
-			prompt.Error("You have no defined credentials in this env")
+		} else if inputParams.provider == "" {
```

I think you should return an error when the user sends an empty provider, so you don't need to check if the provider is empty here.

---

Returning the error makes sense, but we cannot neglect this case. This happens when we prompt the user if she really wants to delete and she says no